File: Backend/TranscomaAPI/Comun/Entidades/Entrada.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace TranscomaAPI.Comun.Entidades
{
    public class Entrada : Entidad
    {
        private DateTime _fechaEntrada;
        private Producto _producto;
        private Almacen _almacen;
        private Cliente _cliente;

        /// <summary>
        /// 
        /// </summary>
        /// <param name="id"></param>
        public Entrada(int id)
        {
            Id = id;
        }

        /// <summary>
        /// Constructor para la consulta de un cliente en especifico
        /// </summary>
        /// <param name="id"></param>
        /// <param name="fechaEntrada"></param>
        /// <param name="producto"></param>
        /// <param name="almacen"></param>
        public Entrada(int id, DateTime fechaEntrada , Producto producto, Almacen almacen)
        {
            Id = id;
            _fechaEntrada = fechaEntrada;
            _producto = producto;
            _almacen = almacen;
        }

        /// <summary>
        /// Contructor para la consulta de todas las entradas
        /// </summary>
        /// <param name="id"></param>
        /// <param name="fechaEntrada"></param>
        /// <param name="producto"></param>
        /// <param name="almacen"></param>
        /// <param name="cliente"></param>
        public Entrada(int id, DateTime fechaEntrada, Producto producto, Almacen almacen, Cliente cliente)
        {
            Id = id;
            _fechaEntrada = fechaEntrada;
            _producto = producto;
            _almacen = almacen;
            _cliente = cliente;
        }

        /// <summary>
        /// Getters y Setters
        /// </summary>
        public DateTime FechaEntrada {get=> _fechaEntrada;set => _fechaEntrada = value;}
        public Producto Producto { get => _producto; set => _producto = value; }
        public Almacen Almacen { get => _almacen; set => _almacen = value; }
        public Cliente Cliente { get => _cliente; set => _cliente = value; }
    }
}

File: frontend/src/app/models/Entrada.ts
import { EntidadBase } from "./EntidadBase";
import { Producto } from "./Producto";
import { Almacen } from "./Almacen";
import { Cliente } from "./Cliente";

export class Entrada extends EntidadBase {

    private _fechaEntrada: Date ;
    private _producto: Producto ;
    private _almacen: Almacen ;
    private _cliente: Cliente ;

    constructor(data?: any) {
        super();
        if (data !== undefined) {
            this.$id = data.id;
            this.$fechaEntrada = data.fechaEntrada;

            if( data.producto != undefined ) {
                this.$producto = new Producto( data.producto );
            }
            else{
                this.$producto = null;
            }

            if( data.almacen != undefined) {
                this.$almacen = new Almacen( data.almacen );
            }
            else{
                this.$almacen = null;
            }

            if( data.cliente != undefined) {
                this.$cliente = new Cliente( data.cliente );
            }
            else{
                this.$cliente = null;
            }
        }
    }

    /** Getters y setters */
    public get $fechaEntrada(): Date {
        return this._fechaEntrada;
    }

    public set $fechaEntrada(value: Date) {
        this._fechaEntrada = value;
    }

    public get $producto(): Producto {
        return this._producto;
    }

    public set $producto(value: Producto) {
        this._producto = value;
    }

    public get $almacen(): Almacen {
        return this._almacen;
    }

    public set $almacen(value: Almacen) {
        this._almacen = value;
    }

    public get $cliente(): Cliente {
        return this._cliente;
    }

    public set $cliente(value: Cliente) {
        this._cliente = value;
    }
}
